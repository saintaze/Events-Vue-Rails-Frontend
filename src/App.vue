<template>
  <div id="app">
    <nav>
      <router-link to="/events" exact >Events</router-link>
      <router-link to="/login">Signin</router-link>
      <router-link to="/about">About</router-link>
    </nav>
    <router-view></router-view>
  </div>
</template>

<script>
import Calendar from './components/Calendar.vue'

export default {
  name: 'App',
  components: {
    Calendar
  }
}
</script>

<style lang="scss">
 html, body {
   padding: 0,;
   margin: 0;
 }
</style>
