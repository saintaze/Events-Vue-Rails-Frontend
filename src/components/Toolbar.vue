<template>
  <header class="header">
    {{brandName}}
  </header>
</template>

<script>
import { BRAND_NAME } from '@/constants.js';

export default {
  name: 'toolbar',
  data(){
    return {
      brandName: BRAND_NAME
    }
  }
}
</script>

<style lang="scss">
  .header {
    height: 40px;
    line-height: 40px;
    padding-left: 1.5em;
    background: #eee;
    border-bottom: 1px solid #ddd;
    font-family: Lucida Grande,Helvetica,Arial,Verdana,sans-serif;
    font-size: 14px;
  }
</style>