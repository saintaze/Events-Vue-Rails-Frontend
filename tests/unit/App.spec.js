import { shallowMount } from "@vue/test-utils"
import App from "@/App.vue"
import Toolbar from '@/components/Toolbar.vue';

let wrapper;

beforeEach(() => {
  wrapper = shallowMount(App, {
    stubs: ['router-view']
  })
})

afterEach(() => {
  wrapper.destroy();
});

describe('App', () => {
  it('should render html correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should contain Toolbar component', () => {
    expect(wrapper.findComponent(Toolbar).exists()).toBe(true)
  });

  
})