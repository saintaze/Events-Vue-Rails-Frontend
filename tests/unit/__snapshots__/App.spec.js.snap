// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render html correctly 1`] = `
<div>
  <toolbar-stub></toolbar-stub>
  <router-view-stub></router-view-stub>
</div>
`;
